processor/processor.go

```diff
@@ -1345,6 +1352,24 @@ func getDiffMetrics(inPU, pu string, inCountMetadataMap map[string]MetricMetadat
 	return diffMetrics
 }
 
+type DupStat struct {
+	Count int
+	Cmp   bool
+}
+
+func IncrementDupStatsMapByKey(m map[string]DupStat, key string, cmp bool, increment int) {
```

Comment:
1. `cmp` should be part of the map's key, i.e. we need a composite key
2. since the map contains values and not pointers, to update the value we should assign it to the map again ([link](https://go.dev/play/p/BA38gHd6tbU))

```go
type dupStatKey struct {
	sourceID    string
	equalSize   bool
}
map[dupStatKey]int
```

Reply:
Done